src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -0,0 +1,217 @@
+{
+    return "* Hourly binding constraints.";
+}
+
+// --- Daily BC constraint ---
+std::string DailyBC::infeasibility()
+{
+    return "Daily BC '" + shortName(splitName()) + "' at day " + timeStep(splitName());
+}
+
+std::string DailyBC::infeasibilityCause()
+{
+    return "* Daily binding constraints,";
+}
+
+// --- Weekly BC constraint ---
+std::string WeeklyBC::infeasibility()
+{
+    return "Weekly BC '" + shortName(splitName());
+}
+
+std::string WeeklyBC::infeasibilityCause()
+{
+    return "* Weekly binding constraints.";
+}
+
+// --- Fictitious load constraint ---
+std::string FictitiousLoad::infeasibility()
+{
+    return "Last resort shedding status at area '" + areaName(splitName()) + "' at hour "
+           + timeStep(splitName());
+}
+
+std::string FictitiousLoad::infeasibilityCause()
+{
+    return "* Last resort shedding status.";
+}
+
+// --- Hydro level constraint ---
+std::string HydroLevel::infeasibility()
+{
+    return "Hydro level constraint at area '" + areaName(splitName()) + "' at hour "
+           + timeStep(splitName());
+}
+
+std::string HydroLevel::infeasibilityCause()
+{
+    return "* Hydro reservoir impossible to manage with cumulative options "
+           "\"hard bounds without heuristic\"";
+}
+
+// --- Short term storage constraint ---
+std::string STS::infeasibility()
+{
+    return "Short-term-storage reservoir constraint at area '" + areaName(splitName())
+           + "' in STS '" + STSname(splitName()) + "' at hour " + timeStep(splitName());
+}
+
+std::string STS::infeasibilityCause()
+{
+    return "* Short-term storage reservoir level impossible to manage. Please check inflows, "
+           "lower & upper curves and initial level (if prescribed),";
+}
+
+// --- Hydro production constraint ---
+std::string HydroProduction::infeasibility()
+{
+    return "Hydro weekly production at area '" + areaName(splitName()) + "'";
+}
+
+std::string HydroProduction::infeasibilityCause()
+{
+    return "* impossible to generate exactly the weekly hydro target";
+}
+
+// --- Constraints factory ---
+std::shared_ptr<WatchedConstraint> createHourlyBC(std::string varName)
+{
+    return std::make_shared<HourlyBC>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createDailyBC(std::string varName)
+{
+    return std::make_shared<DailyBC>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createWeeklyBC(std::string varName)
+{
+    return std::make_shared<WeeklyBC>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createFictitiousLoad(std::string varName)
+{
+    return std::make_shared<FictitiousLoad>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createHydroLevel(std::string varName)
+{
+    return std::make_shared<HydroLevel>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createSTS(std::string varName)
+{
+    return std::make_shared<STS>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createHydroProduction(std::string varName)
+{
+    return std::make_shared<HydroProduction>(varName);
+}
```

Comment:
```suggestion
```

You don't need all this, `std::make_shared<T>` is already a function.

Reply:
Please see [this comment](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2232#discussion_r1672596805)